packages/mobile/src/fiatExchanges/MoonPay.tsx

```diff
@@ -59,6 +62,19 @@ function FiatExchangeWeb({ route }: Props) {
       .catch(() => showError(ErrorMessages.FIREBASE_FAILED)) // Firebase signing function failed
   }, [])
 
+  // XSS vulnerability in Android WebView
+  // https://github.com/react-native-webview/react-native-webview/security/advisories/GHSA-36j3-xxf7-4pqg
+  if (
+    uri &&
+    Platform.OS === 'android' &&
+    version &&
+    compareVersions.compare(version, '83.0.4103.106', '<')
+  ) {
+    navigateToURI(uri)
```

Comment:
This side effect can't happen in render and should be wrapped in a `useEffect` instead.
I think this logic should be moved to where we navigate to this screen instead.